core/thread/src/TReentrantRWLock.cxx

```diff
@@ -263,7 +263,7 @@ TReentrantRWLock<MutexT, RecurseCountsT>::GetStateBefore()
       return nullptr;
    }
 
-   std::unique_ptr<State_t> pState(new State_t);
+   auto pState = new State_t;
```

Comment:
@pcanal please have a look. In principle this is bad because we might add an early return and circumvent the RAII provided by the `unique_ptr`...

Reply:
I guess this change is not ok, since if something happens before the end of the function, `pState` will probably leak. I will remove it from the list that I will merge.